numba/tests/test_asnumbatype.py

```diff
@@ -87,6 +110,22 @@ def test_optional(self):
         self.assertIn("Cannot type Union that is not an Optional",
                       str(raises.exception))
 
+    @unittest.skipIf(sys.version_info < (3, 10), "Requires Python 3.10")
```

Comment:
```suggestion
    @skip_unless_py10_or_later
```
And import `skip_unless_py10_or_later` from `numba.tests.support`